forge/csrc/passes/lower_to_mlir.cpp

```diff
@@ -672,30 +673,55 @@ class MLIRGenerator
         return mlir::FileLineColLoc::get(builder_.getContext(), module.name(), 0, 0);
     }
 
-    /// Get the node location in format "source_location", (graph_id), (node_id)
-    mlir::Location get_node_location(tt::graphlib::Graph *graph, tt::graphlib::Node *node)
+    /// Recursively parses the layer tag by separing on '/', creating a NameLoc for each part and chaining them
+    /// together.
+    ///
+    /// Example desired behaviour (layer -> NameLoc):
+    /// - "module_name/layer_1/layer_2" -> NameLoc("layer_2", NameLoc("layer_1", NameLoc("module_name")))
+    ///
+    mlir::Location parse_layer_tag(std::string_view layer_tag)
+    {
+        auto pos = layer_tag.rfind('/');
+        if (pos != std::string_view::npos)
+        {
+            std::string_view name = layer_tag.substr(pos + 1);
+
+            mlir::StringAttr name_attr = builder_.getStringAttr(name);
+            return mlir::NameLoc::get(name_attr, parse_layer_tag(layer_tag.substr(0, pos)));
+        }
+        else
+        {
+            mlir::StringAttr name_attr = builder_.getStringAttr(layer_tag);
+            return mlir::NameLoc::get(name_attr);
+        }
```

Comment:
Since you are linking this with LLVM, `StringRef` has a lot of very convenient member functions, this could be rewritten as
```suggestion
        StringRef layer_tag;
        auto [prefix, sufix] = layer_tag.rsplit('/');
        if (!sufix.empty())
        {
            mlir::StringAttr name_attr = builder_.getStringAttr(sufix);
            return mlir::NameLoc::get(name_attr, parse_layer_tag(prefix));
        }
        else
        {
            mlir::StringAttr name_attr = builder_.getStringAttr(prefix);
            return mlir::NameLoc::get(name_attr);
        }
```

Reply:
i'll keep the `std` version, just to be aligned with rest of `tt-forge-fe` - we're using the MLIR/LLVM stuff where we need to...